src/NServiceBus.RabbitMQ/Connection/RabbitMqConnectionFactory.cs

```diff
@@ -0,0 +1,46 @@
+﻿namespace NServiceBus.Transports.RabbitMQ.Connection
+{
+    using System;
+    using global::RabbitMQ.Client;
+    using NServiceBus.Transports.RabbitMQ.Config;
+
+    class RabbitMqConnectionFactory
+    {
+        public ConnectionConfiguration Configuration { get; private set; }
+
+        public RabbitMqConnectionFactory(ConnectionConfiguration connectionConfiguration)
+        {
+            if (connectionConfiguration == null)
+            {
+                throw new ArgumentNullException("connectionConfiguration");
```

Comment:
Fair point. BTW, this class is a rename from `ClusterAwareConnectionFactory` so most of the code, including this check, comes from there. I'll re-visit and see if it makes sense to remove/move it.